spring-rest2ts-jackson-js/src/main/java/com/blueveery/springrest2ts/extensions/JacksonJsModelSerializerExtension.java

```diff
@@ -0,0 +1,96 @@
+        objectMapperClass = new TSClass("ObjectMapper", jacksonJSModule, new EmptyImplementationGenerator());
+    }
+
+    @Override
+    public void addComplexTypeUsage(TSComplexElement tsComplexElement) {
+        tsComplexElement.addScopedTypeUsage(objectMapperClass);
+    }
+
+    @Override
+    public void addImplementationSpecificFields(TSComplexElement tsComplexElement) {
+        if (tsComplexElement instanceof TSClass) {
+            TSClass tsParentClass = (TSClass) tsComplexElement;
+            while (tsParentClass.getExtendsClass() != null) {
+                tsParentClass = tsParentClass.getExtendsClass().getReferencedType();
+            }
+            TSField objectMapperField = tsParentClass.getFieldByName(objectMapperFieldName);
+            if (objectMapperField == null) {
+                objectMapperField = new TSField(objectMapperFieldName, tsParentClass, objectMapperClass);
+                objectMapperField.setReadOnly(true);
+                objectMapperField.setInitializationStatement(new TSLiteral("", TypeMapper.tsAny, "new ObjectMapper()"));
+                tsParentClass.addTsField(objectMapperField);
+                tsParentClass.addScopedTypeUsage(objectMapperClass);
+            }
+        }
+    }
+
+    @Override
+    public String generateSerializationCode(String modelVariableName, TSParameter tsParameter) {
+        StringBuilder stringifyStatement = new StringBuilder();
+        if (tsParameter.isOptional() || tsParameter.isNullable()) {
+            stringifyStatement.append(modelVariableName);
+            stringifyStatement.append(" && ");
+        }
+        stringifyStatement.append("this.");
+        stringifyStatement.append(objectMapperFieldName);
+        stringifyStatement.append(".stringify<");
+        stringifyStatement.append(tsParameter.getType().getName());
+        stringifyStatement.append(">(");
+        stringifyStatement.append(modelVariableName);
+        stringifyStatement.append(")");
+        return stringifyStatement.toString();
+    }
+
+    @Override
+    public String generateDeserializationCode(String modelVariableName, TSMethod tsMethod) {
+        try {
+            StringBuilder parseStatement = new StringBuilder();
+            if (tsMethod.isNullable()) {
+                parseStatement.append(modelVariableName);
+                parseStatement.append(" && ");
+            }
+            parseStatement.append("this.");
+            parseStatement.append(objectMapperFieldName);
+            parseStatement.append(".parse<");
+            parseStatement.append(tsMethod.getType().getName());
+            parseStatement.append(">(");
+            parseStatement.append(modelVariableName);
+            parseStatement.append(", {mainCreator: () => ");
+            ILiteral iLiteral = wrapIntoTSLiteralArray(convertToTypeLiteral(tsMethod.getType()));
+            StringWriter stringWriter = new StringWriter();
+            BufferedWriter writer = new BufferedWriter(stringWriter);
```

Comment:
It is better to declare  writers into a try-with-resources block 
```
 try (StringWriter stringWriter = new StringWriter();
       BufferedWriter writer = new BufferedWriter(stringWriter))
        {...}
```

Reply:
in general yes, but  StringWriter has empty close method I don't want to suggest that `close()` do something here